plugin/keyring_vault/vault_curl.cc

```diff
@@ -0,0 +1,177 @@
+#include <algorithm>
+#include "vault_curl.h"
+#include "base64.h"
+
+namespace keyring
+{
+
+static size_t write_response_memory(void *contents, size_t size, size_t nmemb, void *userp)
+{
+  size_t realsize = size * nmemb;
+  std::stringstream *read_data = (std::stringstream *)userp;
+
+  read_data->write((char*)contents, realsize);
+  if (!read_data->good())
+    return 0;
+  return realsize;
+}
+
+std::string Vault_curl::get_error_from_curl(CURLcode curl_code)
+{
+  size_t len = strlen(curl_errbuf);
+  std::stringstream ss;
+  if (curl_code != CURLE_OK)
+  {
+    ss << "Curl returned this error code: " << curl_code;
```

Comment:
uppercased CURL comes from includes from Curl library. My code's curl is lowercase. This makes clear distinction between library code and keyring_vault code. However, if you want me to use CURL everywhere just say "yes" :)